src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -683,9 +682,17 @@
              return;
         }
 
-        var objId = X[self.daoKey_].of.ID.type === 'Long' ? parseInt(this.objId) : this.objId;
-
-        return X[this.daoKey_]
+        var objId, daoKey;
+        if ( self.refObj ) {
+          var res = self.refObj.split(':');
+          daoKey = res[0];
+          objId = X[daoKey].of.ID.type === 'Long' ? parseInt(res[1]) : res[1];
```

Comment:
Same here, as below, showing I do code reviews from the bottom up :(.

Reply:
(: ʞo